pySDC/implementations/problem_classes/odeSystem.py

```diff
@@ -0,0 +1,925 @@
+        Wether or not to use the non-linear form of the problem. The default is False.
+    newton_maxiter : int, optional
+        Maximum number of Newton iteration in solve_system. The default is 200.
+    newton_tol : float, optional
+        Residuum tolerance for Newton iteration in solve_system. The default is 5e-11.
+    stop_at_nan : bool, optional
+        Wheter to stop or not solve_system when getting NAN. The default is True.
+
+    Reference
+    ---------
+    A. Prothero and A. Robinson, On the stability and accuracy of one-step methods for solving
+    stiff systems of ordinary differential equations, Mathematics of Computation, 28 (1974),
+    pp. 145–162.
+    """
+
+    dtype_u = mesh
+    dtype_f = mesh
+
+    def __init__(self, epsilon=1e-3, nonLinear=False, newton_maxiter=200, newton_tol=5e-11, stop_at_nan=True):
+        nvars = 2
+        super().__init__((nvars, None, np.dtype('float64')))
+
+        self.f = self.f_NONLIN if nonLinear else self.f_LIN
+        self.dgInv = self.dgInv_NONLIN if nonLinear else self.dgInv_LIN
+        self._makeAttributeAndRegister(
+            'epsilon', 'nonLinear', 'newton_maxiter', 'newton_tol', 'stop_at_nan', localVars=locals(), readOnly=True
+        )
+        self.work_counters['newton'] = WorkCounter()
+        self.work_counters['rhs'] = WorkCounter()
+
+    # -------------------------------------------------------------------------
+    # g function (analytical solution), and its first and second derivative
+    # -------------------------------------------------------------------------
+    def g(self, t):
+        return np.cos(t)
+
+    def dg(self, t):
+        return -np.sin(t)
+
+    def dg2(self, t):
+        return -np.cos(t)
+
+    # -------------------------------------------------------------------------
+    # f(u,t) and Jacobian functions
+    # -------------------------------------------------------------------------
+    def f(self, u, t):
+        raise NotImplementedError()
+
+    def f_LIN(self, u, t):
+        return -self.epsilon ** (-1) * (u - self.g(t)) + self.dg(t)
+
+    def f_NONLIN(self, u, t):
+        return -self.epsilon ** (-1) * (u**3 - self.g(t) ** 3) + self.dg(t)
+
+    def dgInv(self, u, t):
+        raise NotImplementedError()
+
+    def dgInv_LIN(self, u, t, dt):
+        e = self.epsilon
+        g1, g2 = self.dg(t), self.dg2(t)
+        return np.array([[1 / (dt / e + 1), (dt * g2 + dt * g1 / e) / (dt / e + 1)], [0, 1]])
+
+    def dgInv_NONLIN(self, u, t, dt):
+        e = self.epsilon
+        g, g1, g2 = self.g(t), self.dg(t), self.dg2(t)
+        return np.array(
+            [[1 / (3 * dt * u**2 / e + 1), (dt * g2 + 3 * dt * g**2 * g1 / e) / (3 * dt * u**2 / e + 1)], [0, 1]]
+        )
+
+    # -------------------------------------------------------------------------
+    # pySDC required methods
+    # -------------------------------------------------------------------------
+    def u_exact(self, t, u_init=None, t_init=None):
```

Comment:
Again, the documentation is false and the signature is misleading. There is no call to scipy.

Reply:
Corrected